spring-social-github/src/main/java/org/springframework/social/github/api/impl/AbstractGitHubOperations.java

```diff
@@ -40,9 +42,6 @@ protected void requireAuthorization() {
 	// Using String here instead of URI so I can include braces in the path. See, e.g., RepoTemplate. [WLW]
 	protected String buildUri(String path) {
 //		return URIBuilder.fromUri(API_URL_BASE + path).build();
-		return API_URL_BASE + path;
+		return "https://" + (this.apiHostname != null ? this.apiHostname : "api.github.com") + "/api/v3/" + path;
```

Comment:
Using `UriComponentsBuilder` might be clearer here, plus you get validation and encoding.

Like so:
```suggestion
		return UriComponentsBuilder.newInstance()
                    .scheme("https")
                    .host(this.apiHostname != null ? this.apiHostname : "api.github.com")
                    .path("/api/v3")
                    .path(path)
                    .toUriString();
```

Reply:
Looks like the pattern of doing simple string concat for the URL is used in multiple places, so if we're going to use `UriComponentsBuilder` here, then we should use it everywhere else too. I noticed the logic to check if the hostname is specified or null is repeated. I wonder if we can easily put that logic in one place and just return a `UriComponent` that we can add sub-paths to. The effort required to do that might be more than the benefit. The original code was doing string concat, so I'm fine with status quo. Up to you.